extension/encoding/awslogsencodingextension/internal/unmarshaler/cloudtraillogs/cloudtraillogs.go

```diff
@@ -0,0 +1,308 @@
+	buildInfo component.BuildInfo
+	// Pool the gzip readers, which are expensive to create.
+	gzipPool sync.Pool
+}
+
+var _ plog.Unmarshaler = (*CloudTrailLogsUnmarshaler)(nil)
+
+// CloudTrailRecord represents a CloudTrail log record
+// There is no builtin CloudTrailRecord we can leverage like in S3
+// So we build our own
+type CloudTrailRecord struct {
+	EventVersion                 string         `json:"eventVersion"`
+	EventTime                    string         `json:"eventTime"`
+	EventSource                  string         `json:"eventSource"`
+	EventName                    string         `json:"eventName"`
+	AwsRegion                    string         `json:"awsRegion"`
+	SourceIPAddress              string         `json:"sourceIPAddress"`
+	UserAgent                    string         `json:"userAgent"`
+	RequestID                    string         `json:"requestID"`
+	EventID                      string         `json:"eventID"`
+	EventType                    string         `json:"eventType"`
+	EventCategory                string         `json:"eventCategory"`
+	RecipientAccountID           string         `json:"recipientAccountId"`
+	UserIdentity                 map[string]any `json:"userIdentity"`
+	ResponseElements             map[string]any `json:"responseElements"`
+	RequestParameters            map[string]any `json:"requestParameters"`
+	Resources                    []any          `json:"resources"`
+	ReadOnly                     bool           `json:"readOnly"`
+	ManagementEvent              bool           `json:"managementEvent"`
+	AdditionalEventData          map[string]any `json:"additionalEventData"`
+	TLSDetails                   map[string]any `json:"tlsDetails"`
+	SessionCredentialFromConsole string         `json:"sessionCredentialFromConsole"`
+}
+
+type CloudTrailLogs struct {
+	Records []CloudTrailRecord `json:"Records"`
+}
+
+func NewCloudTrailLogsUnmarshaler(buildInfo component.BuildInfo) plog.Unmarshaler {
+	return &CloudTrailLogsUnmarshaler{
+		buildInfo: buildInfo,
+		gzipPool:  sync.Pool{},
+	}
+}
+
+func (u *CloudTrailLogsUnmarshaler) Unmarshal(buf []byte) (plog.Logs, error) {
+	return u.UnmarshalLogs(buf)
+}
+
+func (u *CloudTrailLogsUnmarshaler) UnmarshalLogs(buf []byte) (plog.Logs, error) {
+	// Check if the content is gzip-compressed
+	var errDecompress error
+	gzipReader, ok := u.gzipPool.Get().(*gzip.Reader)
+	if !ok {
+		gzipReader, errDecompress = gzip.NewReader(bytes.NewReader(buf))
+	} else {
+		errDecompress = gzipReader.Reset(bytes.NewReader(buf))
+	}
+
+	var decompressedBuf []byte
+	if errDecompress == nil {
+		// Content is gzip-compressed, decompress it
+		defer func() {
+			_ = gzipReader.Close()
+			u.gzipPool.Put(gzipReader)
+		}()
+
+		decompressedBuf, errDecompress = io.ReadAll(gzipReader)
+		if errDecompress != nil {
+			return plog.Logs{}, fmt.Errorf("failed to decompress CloudTrail logs: %w", errDecompress)
+		}
+	} else {
+		// Content is not gzip-compressed, use the original buffer
```

Comment:
Can this happen for cloud trail logs? I thought cloud trail logs sent to S3 had to be compressed, unless it was some kind of manual upload, but maybe I am wrong